rfc822/header.go

```diff
@@ -311,6 +311,25 @@ func (h *Header) applyOffset(start *headerEntry, offset int) {
 // SetHeaderValue is a helper method that sets a header value in a message literal.
 // It does not check whether the existing value already exists.
 func SetHeaderValue(literal []byte, key, val string) ([]byte, error) {
+	reader, size, err := SetHeaderValueNoCopy(literal, key, val)
+	if err != nil {
+		return nil, err
+	}
+
+	var b bytes.Buffer
+
+	b.Grow(size)
+
+	if _, err := b.ReadFrom(reader); err != nil {
+		return nil, err
+	}
+
+	return b.Bytes(), nil
+}
+
+// SetHeaderValueNoCopy is a helper method that sets a header value in a message literal.
+// It does not check whether the existing value already exists.
+func SetHeaderValueNoCopy(literal []byte, key, val string) (io.Reader, int, error) {
```

Comment:
Maybe add info to the docstring about what "NoCopy" actually means (which values aren't being copied? What are the behaviour differences between this version and the above version?)

Reply:
Done